src/contracts/PreAMMBatcher.sol

```diff
@@ -404,4 +406,34 @@ contract PreAMMBatcher {
         }
         return newOrders;
     }
+
+    enum Direction {Ascending, Descending}
+
+    function isSortedByLimitPrice(Order[] memory orders, Direction direction)
+        public
+        pure
+        returns (bool)
+    {
+        if (orders.length <= 1) {
+            // All order sets with less than 2 elements are tautologically sorted.
+            return true;
+        }
+        int8 step = 1;
+        if (direction == Direction.Descending) {
+            step = -1;
+        }
+
+        for (uint256 i = 0; i < orders.length - 1; i++) {
+            Order memory orderA = orders[i];
+            Order memory orderB = orders[i + 1];
+            if (
+                int256(orderA.buyAmount.mul(orderB.sellAmount))
```

Comment:
This needs to use `SafeCast`:
```
        int256 step = 1;
        uint256 orderABuyAmount = 1 << 128;
        uint256 orderASellAmount = 1;
        uint256 orderBBuyAmount = 1;
        uint256 orderBSellAmount = (1 << 128) - 1;
        
        return int256(orderABuyAmount.mul(orderBSellAmount))
            .sub(int256(orderBBuyAmount.mul(orderASellAmount)))
            .mul(step);
```

This returns `-340282366920938463463374607431768211457` which means they aren't sorted, but they clearly are since:
`orderABuyAmount * orderBSellAmount > orderBBuyAmount * orderASellAmount`

Reply:
Actually, we can make it overflow a little bit less if we perform arithmetic on the bool:
```
        bool ascending = direction == Direction.Ascending;
        
        uint256 orderABuyAmount = 1 << 128;
        uint256 orderASellAmount = 1;
        uint256 orderBBuyAmount = 1;
        uint256 orderBSellAmount = (1 << 128) - 1;
        
        bool orderAbeforeB = orderABuyAmount.mul(orderBSellAmount) > orderBBuyAmount.mul(orderASellAmount);
        if (orderAbeforeB != ascending) {
            return false;
        }
```

Should there be tests for these overflow cases?

Reply:
I also prefer the bool comparison, it should also save some gas when calling this function.
Extra tests would be great.

Reply:
Unfortunately safe cast isn't implemented for unsigned integers. This would require a PR to OZ.

Reply:
> Unfortunately safe cast isn't implemented for unsigned integers.

Safe-cast isn't needed if we go with the bool comparison in my second example.

Reply:
This suggestion is also nice because we no longer need to SafeCast to Int256 (since everything is now all unsigned)!